### PR TITLE
Fail gracefully if the test project doesn't support test services

### PR DESCRIPTION
**Pull request by pakrym**
@davidfowl

### CODE REVIEW

File: src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/ActivatorUtilities.cs [hunk 1]
```diff
@@ -76,6 +76,9 @@ public static T CreateInstance<T>(IServiceProvider services)
 
             if (constructors.Length == 1)
             {
+                if (services == null)
+                    throw new InvalidOperationException($"Unable to inject into {type.Name} constructor, current host doesn't support dependency injection");
+
```
Comment (rynowak) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/ActivatorUtilities.cs]: This is used in MVC for a bunch of unrelated stuff. This fix isn't close enough to what it's claims to address.

Comment (pakrym) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/ActivatorUtilities.cs]: What else should it address?

**Comment by davidfowl** [comment]
This is why we shouldn't be using the shard code

**Comment by rynowak** [comment]
IMO testhost should just have the same semantics and limitations as DNX as a whole.

**Comment by davidfowl** [comment]
Nope it should just have its own entry point

**Comment by pakrym** [comment]
@rynowak, I can change error message to something like "Requesting code does not provide dependency injection services". It would be semantically correct for all use-cases

**Comment by pakrym** [comment]
:up: :date:, any conclusion?

File: src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/ActivatorUtilities.cs [hunk 2]
```diff
@@ -79,6 +79,9 @@ public static T CreateInstance<T>(IServiceProvider services)
                 ParameterInfo[] parameters = constructors[0].GetParameters();
                 return services =>
                 {
+                    if (services == null)
+                        throw new InvalidOperationException($"Unable to inject into {type.Name} constructor, current host doesn't support dependency injection");
```
Comment (victorhurdugaci) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/ActivatorUtilities.cs]: Always use brackets around if's body

Comment (victorhurdugaci) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/ActivatorUtilities.cs]: Suggestion for message: "Cannot create instance of {type} because the current host doesn't support dependency injection"

Comment (pakrym) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/ActivatorUtilities.cs]: It looks like it's not only used in hosts, anyway there is no conclusion if we need this at all.

**Comment by pakrym** [comment]
I guess we don't care anymore about this.
